### DISCUSSION TITLE
Self-hosting configuration questions

### DISCUSSION THREAD

**Discussion by vanchaxy** [anyproto/contributors#28] [Self-hosting]
Greetings! I'm looking to create a Docker Compose setup for local development and a Helm chart for Kubernetes deployment. At the moment, I have a few questions:

1. Is it possible to have multiple coordinators? If so, does it make sense to do so?
2. I noticed that Docker images in GitHub workflows are based on Ubuntu, which tends to be quite large. Is there a specific reason for this choice?
3. Can I use a URL as an address definition? For instance, can I use "controller.domain.com:8080" instead of "192.168.0.1:8080"?
4. If the coordinator is responsible for fetching a node's address, does it imply that both the coordinator and the node should be reachable from the client or other clients in the same network? Would I need to expose all nodes to the public if I want to sync from anywhere?

Thank you for your help!

**Comment by cheggaaa** [comment]
Hi

1. It's possible but does not make sense for self-host purposes
2. Any fresh Linux should be ok
3. Yes, you can use DNS names as well
4. Yes. Coordinator responsibilities are:
    - provide network configuration updates
    - provide storage size limiting 
    - create permits for clients to push their data to any-sync-nodes
    - some functional for deletion spaces (it's probably will be moved to the acl system)

If you want to deploy your network in public you should understand that anyone who knows the address can use it. It's the responsibility of the coordinator to provide access control and it's implemented on a protocol level. But in the current implementation, we haven't any rules or filters (e.g. by IP or password/key) yet.
But of course, you can restrict access to coordinators on the system level for example with a firewall.

**Reply by vanchaxy** [reply]
Thanks for the answer!
